quickjs.c

```diff
@@ -1927,7 +1944,7 @@ void JS_FreeRuntime(JSRuntime *rt)
 
 #ifdef DUMP_LEAKS
     /* leaking objects */
-    {
+    if ((rt->dump_flags & DUMP_LEAKS_FLAG) == DUMP_LEAKS_FLAG) {
```

Comment:
All of these ones that only really affect FreeRuntime could be without an ifdef IMHO since there won't be a bug penalty...

Reply:
The `ifdef` is used to avoid compiling and linking the dump code if not required du reduce the library size.  This is useful for embedded targets and __wasi__.

Reply:
Maybe some ENABLE_DUMPING (or a better name) which gates them all would be good?

Reply:
Actually, I found a much simple way:  if the `DUMP_XXX` is defined as nothing or 0, output is unconditional, if it is defined as a bit value, output is conditional on `rt->dump_flag & DUMP_XXX`, set via the command line option.  The `DUMP_XXX_FLAG` definitions are gone and the `CMakefile` unchanged.  Commit pending.